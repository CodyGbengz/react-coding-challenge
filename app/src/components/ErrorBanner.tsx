import React from 'react';
import styled from 'styled-components';

export interface ErrorBannerProps {
    message: string;
}

const StyledDiv = styled.div`
    background: #f00;
    color: #fff;
    height: 5%;
    padding: 5px;
    position: absolute;
    text-align: center;
    top: 0px;
    width: 92.5%;
`

export const ErrorBanner: React.FC<ErrorBannerProps> = ({ message }) => {
    return (
        <StyledDiv>
            {message}
        </StyledDiv>
    )
}