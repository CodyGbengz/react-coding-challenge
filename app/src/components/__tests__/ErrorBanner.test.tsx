
import React from 'react';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom'

import { ErrorBanner } from '../ErrorBanner';


describe('ErrorBanner', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ErrorBanner message={'this is an error'}/>
        )
        expect(screen.getByText('this is an error')).toBeInTheDocument();
        expect(container.parentElement).toMatchSnapshot();
    })
})