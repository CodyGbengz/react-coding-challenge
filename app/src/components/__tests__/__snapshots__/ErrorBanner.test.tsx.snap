// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBanner should render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdvvtL bxgYUc"
    >
      this is an error
    </div>
  </div>
</body>
`;
